python/Ganga/new_tests/GPI/GangaList/TestCopy.py

```diff
@@ -43,7 +36,7 @@ def testCopy(self):
         gl2 = copy.copy(gl)
         assert len(gl2) == len(gl), 'lists must be equal'
         assert gl2 is not gl, 'list must be copies'
-        #assert gl[0] is gl2[0], 'the references must be copied' ## rcurrie 09/12/15 Not sure this is valid given we have to cope with GangaObjects here which may not be copyable!
+        assert gl[0] is gl2[0], 'the references must be copied'
```

Comment:
This is up for debate but my personal feeling is that this test is actually for under-the-hood stuff and so keeping the references is the correct and expected behaviour. As regards the user, they should never knowingly manipulate GangaLists anyway and when the set do a copy through the GPI, the GangaList should (and does I believe) do clones of the contained objects thanks to the proxy layer. However, in the Ganga code, I think it makes more sense for GangaList to behave like a python list. In addition, this was the behaviour before in 6.0.44 and earlier.